src/many/src/cose_helpers.rs

```diff
@@ -5,34 +5,84 @@ use coset::{
 };
 use std::collections::{BTreeMap, BTreeSet};
 
+/// Build an EdDSA CoseKey
+///
+/// # Arguments
+///
+/// * `x` - Public key
+/// * `d` - Private key
+pub fn eddsa_cose_key(x: Option<Value>, d: Option<Value>) -> CoseKey {
+    // Allocate at least the public key
+    let mut params: Vec<(Label, Value)> = Vec::with_capacity(1);
+    let mut key_ops: BTreeSet<KeyOperation> = BTreeSet::new();
+
+    if let Some(x) = x {
+        params.push((Label::Int(coset::iana::OkpKeyParameter::X as i64), x));
+        key_ops.insert(KeyOperation::Assigned(coset::iana::KeyOperation::Verify));
+    }
+
+    if let Some(d) = d {
+        params.push((Label::Int(coset::iana::OkpKeyParameter::D as i64), d));
+        key_ops.insert(KeyOperation::Assigned(coset::iana::KeyOperation::Sign));
+    }
+
+    // The CoseKeyBuilder is too limited to be used here
+    CoseKey {
+        kty: KeyType::Assigned(coset::iana::KeyType::OKP),
+        alg: Some(Algorithm::Assigned(coset::iana::Algorithm::EdDSA)),
+        key_ops,
+        params: [
+            vec![(
+                Label::Int(coset::iana::OkpKeyParameter::Crv as i64),
+                Value::from(coset::iana::EllipticCurve::Ed25519 as u64),
+            )],
+            params,
+        ]
+        .concat(),
+        ..Default::default()
+    }
+}
+
+/// Build an ECDSA CoseKey
+///
+/// # Arguments
+///
+/// * `(x, y)` - Public key
+/// * `d` - Private key
+pub fn ecdsa_cose_key((x, y): (Option<Value>, Option<Value>), d: Option<Value>) -> CoseKey {
```

Comment:
Same as above; `(x, y)` shouldn't be optional.

Reply:
Fixed